{
  if (result != (exception ? 0 : VALUE) || enter != VALUE * 2) {
    throw new AssertionError();
  }
  exit++;
  return VALUE;
}
